provider/packet/provider.go

```diff
@@ -0,0 +1,137 @@
+package packet
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/juju/errors"
+	"github.com/juju/jsonschema"
+	"github.com/juju/juju/environs"
+	environscloudspec "github.com/juju/juju/environs/cloudspec"
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/environs/context"
+	"github.com/lxc/lxd/shared/logger"
+	"github.com/packethost/packngo"
+)
+
+type environProvider struct {
+	environProviderCredentials
+}
+
```

Comment:
```suggestion
// CloudSchema returns the schema used to validate input for add-cloud.  Since
// this provider does not support custom clouds, this always returns nil.
```